packages/blade/src/components/Drawer/__tests__/Drawer.web.test.tsx

```diff
@@ -0,0 +1,116 @@
+/* eslint-disable @typescript-eslint/no-empty-function */
+import userEvents from '@testing-library/user-event';
+import { waitFor } from '@testing-library/react';
+import React from 'react';
+import type { DrawerProps } from '../';
+import { Drawer, DrawerBody, DrawerHeader } from '../';
+import renderWithTheme from '~utils/testing/renderWithTheme.web';
+import { Badge } from '~components/Badge';
+import { Button } from '~components/Button';
+import { Text } from '~components/Typography';
+import { AnnouncementIcon, DownloadIcon } from '~components/Icons';
+
+const BasicDrawer = (props: Partial<DrawerProps>): React.ReactElement => {
+  const [isOpen, setIsOpen] = React.useState(false);
+  return (
+    <>
+      <Drawer
+        {...props}
+        isOpen={isOpen}
+        onDismiss={() => {
+          setIsOpen(false);
+          props.onDismiss?.();
+        }}
+        accessibilityLabel="Test Drawer"
+      >
+        <DrawerHeader title="Drawer" />
+        <DrawerBody>
+          <Text>Test Content</Text>
+        </DrawerBody>
+      </Drawer>
+      <Button onClick={() => setIsOpen(true)}>Toggle Drawer</Button>
+    </>
+  );
+};
+
+describe('Drawer', () => {
+  it('renders a Drawer', () => {
+    const { getByRole } = renderWithTheme(
+      <Drawer isOpen={true} onDismiss={() => {}} accessibilityLabel="Test Drawer">
+        <DrawerHeader
+          leading={<AnnouncementIcon size="large" />}
+          title="Address Details"
+          subtitle="Saving addresses will improve your checkout experience"
+          trailing={<Button icon={DownloadIcon} />}
+          titleSuffix={<Badge color="positive">NEW</Badge>}
+        />
+        <DrawerBody>
+          <Text>Test Content</Text>
+        </DrawerBody>
+      </Drawer>,
+    );
+    expect(getByRole('dialog')).toMatchSnapshot();
+  });
+
+  it('should open and close drawer', async () => {
+    const user = userEvents.setup();
+
+    const onDismiss = jest.fn();
+
+    const { getByRole, queryByRole } = renderWithTheme(<BasicDrawer onDismiss={onDismiss} />);
+    await user.click(getByRole('button', { name: 'Toggle Drawer' }));
+    await waitFor(() => expect(getByRole('dialog')).toBeVisible());
+    await user.click(getByRole('button', { name: 'Close' }));
+    await waitFor(() => expect(queryByRole('dialog')).not.toBeInTheDocument());
+    expect(onDismiss).toHaveBeenCalledTimes(1);
```

Comment:
We can also test this in interaction tests, see carousel interaction tests

Reply:
Not sure why but its not working as expected in interaction test. Like the onDismissMock is getting called but something going wrong with referrencing variable from global scope I guess. The onDismissMock reference getting called might be different from onDiss

<img width="957" alt="image" src="https://github.com/razorpay/blade/assets/30949385/2b5b5568-6388-4947-8643-0a939d8d8bdc">

Anyways I think this should be tested on unit test since on a unit you're checking if your prop and handlers work or not.

On interaction I have added state handling that eventually ensures drawer is getting closed or not when the close state is handled